packages/graphql/tests/integration/issues/440.int.test.ts

```diff
@@ -0,0 +1,111 @@
+import neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+
+describe("https://github.com/neo4j/graphql/issues/440", () => {
+    let driver: Driver;
+    const typeDefs = gql`
+        type Video {
+            id: ID! @id(autogenerate: false)
+            categories: [Category!]! @relationship(type: "IS_CATEGORIZED_AS", direction: OUT)
+        }
+
+        type Category {
+            id: ID! @id(autogenerate: false)
+            videos: [Video!]! @relationship(type: "IS_CATEGORIZED_AS", direction: IN)
+        }
+    `;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    test("should be able to disconnect 2 nodes while creating one in the same mutation", async () => {
+        const session = driver.session();
+        const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+        const videoID = generate({ charset: "alphabetic" });
+        const catIDs = Array(3).fill(0).map(() => generate({ charset: "alphabetic" }));
+
+        await session.run(
+            `CREATE (v:Video {id: $videoID}),
+                (v)-[:IS_CATEGORIZED_AS]->(:Category {id: $c0}),
+                (v)-[:IS_CATEGORIZED_AS]->(:Category {id: $c1})`,
+            { videoID, c0: catIDs[0], c1: catIDs[1] }
+        );
+
+        const variableValues = {
+            id: videoID,
+            fields: {
+                categories: [{
+                    disconnect: [{
+                        where: {
+                            node: { id_IN: [catIDs[0], catIDs[1]] }
+                        }
+                    }],
+                    create: [
+                        { node: { id: catIDs[2] } }
+                    ]
+                }]
+            }
+        };
+
+        const mutation = `
+            mutation updateVideos($id: ID!, $fields: VideoUpdateInput!) {
+                updateVideos(where: {id: $id}, update: $fields) {
+                    videos {
+                        id
+                        categories {
+                            id
+                        }
+                    }
+                }
+            }
+        `;
+
+        try {
+            await neoSchema.checkNeo4jCompat();
+
+            const mutationResult = await graphql({
+                schema: neoSchema.schema,
+                source: mutation,
+                contextValue: { driver, driverConfig: { bookmarks: [session.lastBookmark()] } },
+                variableValues,
+            });
+
+            expect(mutationResult.errors).toBeFalsy();
+
+            expect(mutationResult?.data?.updateVideos?.videos).toHaveLength(1);
+            expect(mutationResult?.data?.updateVideos?.videos[0].id).toEqual(videoID);
+            expect(mutationResult?.data?.updateVideos?.videos[0].categories).toHaveLength(1);
+            expect(mutationResult?.data?.updateVideos?.videos[0].categories[0].id).toEqual(catIDs[2]);
```

Comment:
Could you add in a Cypher match for category with ID `catIDs[2]` down here? Sometimes, the GraphQL API doesn't show if "extra" nodes have been created, and this would give us better confidence that only one category has been created instead of two as before this fix. 

```ts
const categories = await session.run(`MATCH (c:Category {id: $id) RETURN c`, { id: catIDs[2] });
expect(categories.records).toHaveLength(2);
```

Reply:
As this was a very much "belts and braces" request and I'm confident that the solution is good, resolving and approving so we can get merged and released.

Reply:
Sorry for the time before a response, I was quite busy yesterday :confused:

Reply:
No need to apologise, this project is our job, not yours, we really appreciate the contribution! 

Don't bother with this now, we can add ourselves if we feel we need to. 🙂